### PR TITLE
CR-1090661 xbutil2 validate fails on U250 2RP shell

### PR DESCRIPTION
**Pull request by sadasivanshaiju**

### CODE REVIEW

**Comment by sadasivanshaiju** [comment]
Here is the test result with the patch

root@pranjal-dell:/proj/rdi/staff/sadasiva/XRT/build/Release# xbmgmt partition --program --name xilinx_u250_gen3x16_xdma_shell_4_1
Programming PLP on Card [0000:17:00.0]...
Partition file: /opt/xilinx/firmware/u250/gen3x16/xdma-shell/partition.xsabin
Program successfully
root@pranjal-dell:/proj/rdi/staff/sadasiva/XRT/build/Release# xbutil --new  validate 
-----------------------------------------------------
 Invoking next generation of the xbutil application 
-----------------------------------------------------
Starting validation for 1 devices

Validate device [0000:17:00.1]
Platform            : xilinx_u250_gen3x16_xdma_shell_4_1
SC Version          : 4.6.11
Platform ID         : 0x0
-------------------------------------------------------------------------------
Test 1 [0000:17:00.1]       : Aux connection 
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 2 [0000:17:00.1]       : PCIE link 
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 3 [0000:17:00.1]       : SC version 
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 4 [0000:17:00.1]       : Verify kernel 
[PASSED] : Test duration: < 3s >
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 5 [0000:17:00.1]       : DMA 
    Details                 : Host -> PCIe -> FPGA write bandwidth = 9415.702339 MB/s
                              Host <- PCIe <- FPGA read bandwidth = 11781.490174 MB/s
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 6 [0000:17:00.1]       : iops 
[PASSED] : Test duration: < 1s >
    Details                 : IOPS: 227851 (hello)
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 7 [0000:17:00.1]       : Bandwidth kernel 
[PASSED] : Test duration: < 1m 16s >
    Details                 : Maximum throughput: 52502 MB/s
    Test Status             : [PASSED]
-------------------------------------------------------------------------------
Test 8 [0000:17:00.1]       : vcu 
Validation completed
Validation Summary
------------------
1  device(s) evaluated
1  device(s) validated successfully
0  device(s) had exceptions during validation

Validated successfully [1 device(s)]
  - [0000:17:00.1] : xilinx_u250_gen3x16_xdma_shell_4_1

Validation Exceptions [0 device(s)]

Warnings produced during test [0 device(s)] (Note: The given test successfully validated)
root@pranjal-dell:/proj/rdi/staff/sadasiva/XRT/build/Release#

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 1]
```diff
@@ -857,6 +857,12 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
 
     std::stringstream run_details;
     size_t block_size = 16 * 1024 * 1024; // Default block size 16MB
+
+   // check if the bank has enough memory to allocate
```
Comment (xdavidz) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:861]: @AShivangi I am not sure we accept this type of comment, should we use /* for multiple lines?

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 2]
```diff
@@ -857,6 +857,12 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
...
+   // m_size is in KB so convert block_ize (bytes) to KB for comparision
+   if(mem.m_size < (block_size/1024))
+             continue;
```
Comment (xdavidz) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:864]: indent issue.

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 3]
```diff
@@ -857,6 +857,12 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
 
     std::stringstream run_details;
     size_t block_size = 16 * 1024 * 1024; // Default block size 16MB
+
+   // check if the bank has enough memory to allocate
+   // m_size is in KB so convert block_ize (bytes) to KB for comparision
```
Comment (xdavidz) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:862]: I suppose the xbutil1 checks this condition but xbutil2 doesn't, therefore there is an regression? Could you please explain the root cause of the fix and add into the comments of the CR?

**Review by xdavidz — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp (hunk 3 above)
Comment (sadasivanshaiju) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp]: The memory bank got initially didn't have enough memory  for  the DMA test . So continue the loop till you get enough memory.

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp]: No, according to C++ standards, we should have "//" comments even for multiple lines

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 4]
```diff
@@ -857,6 +857,13 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
 
     std::stringstream run_details;
     size_t block_size = 16 * 1024 * 1024; // Default block size 16MB
+
+   /* check if the bank has enough memory to allocate
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:861]: please change these back to //

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 5]
```diff
@@ -857,6 +857,13 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
...
+     m_size is in KB so convert block_ize (bytes) to KB for comparision */
```
Comment (AShivangi) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:862]: _block_size_

**Review by AShivangi — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 6]
```diff
@@ -857,6 +857,13 @@ dmaTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
...
+
+  if (mem.m_size < (block_size/1024))
+    continue;
+
```
Comment (xdavidz) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:866]: /* check if the bank has enough memory to allocate
     m_size is in KB so convert block_ize (bytes) to KB for comparision */

  if (mem.m_size < (block_size/1024))
    continue;

Still has indent issue. and the comment should be 
/*
 * check if the bank has enough memory to allocate
 * ...
 * /

Comment (sadasivanshaiju) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:866]: I am not clear . 4 space in front of which line

Comment (xdavidz) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:866]: we can open the raw source file here: https://github.com/Xilinx/XRT/blob/401b1c7d3cdd9bd0376f7fcb574a2dea56e839c6/src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

then goto line 860

we can see the previous line858, 859 has 4 space in front of each code
on line 861, it seems there is 3 spaces (expect 4 in align with 858, 859)
on line 862, it seems there is 2 spaces (expect 4)
on line 864, it seems there is 2 spaces (expect 4)
on line 865, it seems there is 4 spaces (expect 6)

**Review by xdavidz — APPROVED** [review]

**Review by AShivangi — APPROVED** [review]

**Pull request merged**